runtime/gc_ctrl.c

```diff
@@ -511,7 +511,6 @@ CAMLprim value caml_gc_minor(value v)
   CAMLassert (v == Val_unit);
   caml_request_minor_gc ();
   caml_gc_dispatch ();
-  caml_check_urgent_gc (Val_unit);
```

Comment:
The minor collection can trigger new finalisers. We can call `caml_process_pending_callbacks` here.

Reply:
Even better, I think we should. I just now see that the call to finalisers used to be in gc_dispatch. I thought that the old version did not call finalisers (my bad).

I do not want to be annoying, but I have to ask: caml_gc_dispatch is in the public interface, but on the other hand there is no occurrence in opam packages, so is there a reason that makes that it is not supposed to be called from user programs? If not, should the Changes files not mention the changing behaviour? I have the impression that it is public "by accident" (somehow nothing is internal in minor_gc.h, the latter of which is included in no opam package). (However, this also changes functions like `caml_minor_collection` which used to call finalisers and now don't, same remark for `caml_ephemeron_get_*_copy`...)

Reply:
Yes, my interpretation is that everything in `minor_gc.h` is private. I don't think there is any point in making this change explicit in `Changes`.

Reply:
I agree with the interpretation. I think it could be clearer to the user though by making them internal (but, too late).